### PR TITLE
add user roles

### PR DESCRIPTION
**Pull request by Thiblef**

### CODE REVIEW

File: UserAdminBundle/DependencyInjection/Compiler/RoleCompilerPass.php [hunk 1]
```diff
@@ -22,6 +22,10 @@ public function process(ContainerBuilder $container)
     {
         $this->addRoles($container, array(
             AdministrationPanelStrategy::ROLE_ACCESS_USER,
+            AdministrationPanelStrategy::ROLE_ACCESS_CREATE_USER,
+            AdministrationPanelStrategy::ROLE_ACCESS_UPDATE_USER,
+            AdministrationPanelStrategy::ROLE_ACCESS_DELETE_USER,
+
```
Comment (itkg-alavieille) [inline UserAdminBundle/DependencyInjection/Compiler/RoleCompilerPass.php]: remove this line

File: UserAdminBundle/Tests/Functional/Controller/ApiControllersSecurityTest.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+namespace OpenOrchestra\UserAdminBundle\Tests\Functional\Controller;
+
+use OpenOrchestra\ApiBundle\Tests\Functional\Controller\AbstractControllerTest;
+
+/**
+ * Class ApiControllersSecurityTest
+ */
+class ApiControllersSecurityTest extends AbstractControllerTest
+{
+    protected $username = "userLog";
```
Comment (itkg-alavieille) [inline UserAdminBundle/Tests/Functional/Controller/ApiControllersSecurityTest.php]: when this PR https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1152 is merge, use this use : userNoAccess

File: UserAdminBundle/Tests/Functional/Controller/FormControllersSecurityTest.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+namespace OpenOrchestra\UserAdminBundle\Tests\Functional\Controller;
+
+use OpenOrchestra\BackofficeBundle\Tests\Functional\Controller\AbstractControllerTest;
+
+/**
+ * Class FormControllersSecurityTest
+ */
+class FormControllersSecurityTest extends AbstractControllerTest
+{
+    protected $username = 'userLog';
```
Comment (itkg-alavieille) [inline UserAdminBundle/Tests/Functional/Controller/FormControllersSecurityTest.php]: when this PR https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1152 is merge, use this use : userNoAccess

File: UserAdminBundle/Transformer/UserCollectionTransformer.php [hunk 4]
```diff
@@ -3,14 +3,16 @@
 namespace OpenOrchestra\UserAdminBundle\Transformer;
 
 use Doctrine\Common\Collections\Collection;
+use OpenOrchestra\Backoffice\NavigationPanel\Strategies\AdministrationPanelStrategy;
 use OpenOrchestra\BaseApi\Facade\FacadeInterface;
+use OpenOrchestra\BaseApi\Transformer\AbstractSecurityCheckerAwareTransformer;
 use OpenOrchestra\BaseApi\Transformer\AbstractTransformer;
```
Comment (itkg-alavieille) [inline UserAdminBundle/Transformer/UserCollectionTransformer.php]: remove this use

Comment (itkg-alavieille) [inline UserAdminBundle/Transformer/UserCollectionTransformer.php]: remove this unused use OpenOrchestra\BaseApi\Transformer\AbstractTransformer;

File: UserAdminBundle/Transformer/UserTransformer.php [hunk 5]
```diff
@@ -2,7 +2,9 @@
 
 namespace OpenOrchestra\UserAdminBundle\Transformer;
 
+use OpenOrchestra\Backoffice\NavigationPanel\Strategies\AdministrationPanelStrategy;
 use OpenOrchestra\BaseApi\Facade\FacadeInterface;
+use OpenOrchestra\BaseApi\Transformer\AbstractSecurityCheckerAwareTransformer;
 use OpenOrchestra\BaseApi\Transformer\AbstractTransformer;
```
Comment (itkg-alavieille) [inline UserAdminBundle/Transformer/UserTransformer.php]: remove this unused use OpenOrchestra\BaseApi\Transformer\AbstractTransformer;

File: UserAdminBundle/Resources/translations/role.en.yml [hunk 6]
```diff
@@ -1,2 +1,5 @@
 open_orchestra_role:
-    role_access_user: Manage users
+    role_access_user: Manage user
+    role_access_create_user: Create user
```
Comment (itkg-ngilain) [inline UserAdminBundle/Resources/translations/role.en.yml]: a user

File: UserAdminBundle/Transformer/UserTransformer.php [hunk 7]
```diff
@@ -23,8 +26,9 @@ class UserTransformer extends AbstractTransformer
      * @param EventDispatcherInterface $eventDispatcher
      * @param TranslationChoiceManager $translationChoiceManager
      */
-    public function __construct(EventDispatcherInterface $eventDispatcher, TranslationChoiceManager $translationChoiceManager)
+    public function __construct(EventDispatcherInterface $eventDispatcher, TranslationChoiceManager $translationChoiceManager, AuthorizationCheckerInterface $authorizationChecker)
```
Comment (itkg-ngilain) [inline UserAdminBundle/Transformer/UserTransformer.php]: Can you put your parameters on several lines, this one is too long

**Comment by Thiblef** [comment]
@itkg-ngilain @itkg-alavieille

File: UserAdminBundle/Tests/Functional/Controller/ApiControllersSecurityTest.php [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    /**
+     * @return array
+     */
+    public function provideApiUrl()
+    {
+        return array(
+            array('/api/user/root'),
+            array('/api/user'),
+            array('/api/user/root/delete', 'DELETE'),
+
```
Comment (itkg-alavieille) [inline UserAdminBundle/Tests/Functional/Controller/ApiControllersSecurityTest.php]: remove this empty lien

File: UserAdminBundle/Tests/Functional/Controller/FormControllersSecurityTest.php [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    /**
+     * @return array
+     */
+    public function provideApiUrl()
+    {
+        return array(
+            array('/admin/user/new'),
+            array('/admin/user/form/root'),
+            array('/admin/user/password/change/root'),
+
```
Comment (itkg-alavieille) [inline UserAdminBundle/Tests/Functional/Controller/FormControllersSecurityTest.php]: remove this empty line

**Comment by itkg-alavieille** [comment]
else empty line :+1:

**Pull request merged**